### PR TITLE
Change event processor handlers from function to consumer

### PR DESCRIPTION
**Pull request by srnagar**
This change is to update the event handlers in EventProcessorBuilder to use `Consumer` instead of `Function`.  This is based on the API review feedback. Other feedback from API review will be addressed in follow-up PRs.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/README.md [hunk 1]
```diff
@@ -308,7 +308,9 @@ class Program {
             .processEvent(partitionEvent -> {
                 System.out.println("Partition id = " + partitionEvent.getPartitionContext().getPartitionId() + " and "
                     + "sequence number of event = " + partitionEvent.getEventData().getSequenceNumber());
-                return Mono.empty();
+            })
+            .processError(errorContext -> {
+                System.out.println("Error occured while processing events " + errorContext.getThrowable().getMessage);
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/README.md]: Does getMessage work? Surprised it's not a function.

File: sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventProcessorSample.java [hunk 2]
```diff
@@ -4,12 +4,11 @@
 package com.azure.messaging.eventhubs;
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventProcessorSample.java]: One of the feedbacks was to have a reference sample of how we expect users implement handle errors (ie. if it is transient, or an AmqpException or etc). Is this in another PR?

File: sdk/eventhubs/azure-messaging-eventhubs/README.md (hunk 1 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/README.md]: Oh it's a typo. Fixed it.

File: sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventProcessorSample.java (hunk 2 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/EventProcessorSample.java]: I will add that in another PR.

**Pull request merged**